app/src/main/java/edu/grinnell/grinnell_publications_android/Services/Implementation/RealmLocalClient.java

```diff
@@ -49,22 +71,43 @@ public void savePublication(Publication publication) {
 
     @Override
     public List<Publication> getAllPublications() {
-        return null;
+        Realm realm = Realm.getDefaultInstance();
+        RealmResults<RealmPublication> pubResults = realm.where(RealmPublication.class).findAll();
+        List<Publication> mResults = new ArrayList<>();
+        mResults.addAll(pubResults.subList(0, pubResults.size()));
+        return mResults;
     }
 
     @Override
     public List<Story> getAllStories(int page, int numStoriesPerPage) {
-        return null;
+        Realm realm = Realm.getDefaultInstance();
+        RealmResults<RealmStory> mRealmResults= realm.where(RealmStory.class).findAll();
+        List<Story> mResults = new ArrayList<>();
+        mResults.addAll(mRealmResults.subList(0, mRealmResults.size())); //puts realmResults in a list
+        return mResults;
     }
 
     @Override
     public List<Story> getRecentStories(List<Integer> subscribedPublicationIds, Date mostRecentStory) {
-        return null;
+        Realm realm = Realm.getDefaultInstance();
+        List<Story> allStories = new ArrayList<>();
+
+        for (Integer currentPublicationId: subscribedPublicationIds) { //for each subscribed publication
+            RealmResults<RealmStory> publicationStories = realm
+                    .where(RealmStory.class)
+                    .equalTo(CONSTANTS.PUBLICATIONID, currentPublicationId)
```

Comment:
You should use static imports to shorten this code so you're just calling `PUBLICATION_ID`